src/main/kotlin/org/pkl/lsp/completion/UnqualifiedAccessCompletionProvider.kt

```diff
@@ -0,0 +1,265 @@
+/**
+ * Copyright © 2024 Apple Inc. and the Pkl project authors. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.pkl.lsp.completion
+
+import org.eclipse.lsp4j.CompletionItem
+import org.eclipse.lsp4j.CompletionItemKind
+import org.eclipse.lsp4j.CompletionParams
+import org.eclipse.lsp4j.InsertTextFormat
+import org.pkl.lsp.PklBaseModule
+import org.pkl.lsp.Project
+import org.pkl.lsp.ast.*
+import org.pkl.lsp.decapitalized
+import org.pkl.lsp.packages.dto.PklProject
+import org.pkl.lsp.resolvers.ResolveVisitors
+import org.pkl.lsp.resolvers.Resolvers
+import org.pkl.lsp.resolvers.withoutShadowedElements
+import org.pkl.lsp.type.Type
+import org.pkl.lsp.type.computeThisType
+import org.pkl.lsp.type.inferExprTypeFromContext
+
+class UnqualifiedAccessCompletionProvider(private val project: Project) : CompletionProvider {
+
+  override fun getCompletions(
+    node: PklNode,
+    params: CompletionParams,
+    collector: MutableList<CompletionItem>,
+  ) {
+    val line = params.position.line + 1
+    val column = params.position.character + 1
+    val context = node.containingFile.pklProject
+    val base = project.pklBaseModule
+    val actualNode =
+      node.enclosingModule?.findBySpan(line, column) as? PklUnqualifiedAccessExpr ?: return
```

Comment:
I think we'll need something a little more sophisticated, because it can't handle cases like:

```
foo = <cursor>
```

```
foo {
  <cursor>
}
```

```
foo = let (bar = <cursor>) bar + 5
```

Maybe one approach here could be to: inject an expression, then see if the cursor position becomes unqualified access?